buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/UpdateIntellijVersions.kt

```diff
@@ -0,0 +1,90 @@
+package com.virtuslab.gitmachete.buildsrc
+
+import com.virtuslab.gitmachete.buildsrc.IntellijVersionHelper.getMajorPart
+import com.virtuslab.gitmachete.buildsrc.IntellijVersionHelper.toBuildNumber
+import com.virtuslab.gitmachete.buildsrc.IntellijVersionHelper.versionIsNewerThan
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+import org.jsoup.Jsoup
+
+open class UpdateIntellijVersions : DefaultTask() {
+  @Internal
+  val intellijReleasesUrl: String = "https://www.jetbrains.com/intellij-repository/releases/"
+
+  private fun findMatchingVersionNewerThan(regex: Regex, thresholdVersion: String): String? {
+    val htmlContent = Jsoup.connect(intellijReleasesUrl).get()
+    val links = htmlContent.select(
+      "a[href^=${intellijReleasesUrl}com/jetbrains/intellij/idea/BUILD/][href$=.txt]"
+    )
+
+    for (link in links) {
+      val attr = link.attr("href")
+      val matchResult = regex.find(attr)
+
+      if (matchResult != null) {
+        val foundVersion = matchResult.value
+
+        if (foundVersion versionIsNewerThan thresholdVersion) {
+          return foundVersion
+        }
+      }
+    }
+    return null
+  }
+  private fun findReleaseNewerThan(latestStable: String): String? {
+    return findMatchingVersionNewerThan(Regex("(?<=BUILD-)(\\d+\\.)+\\d+(?=.txt)"), latestStable)
+  }
+
+  private fun findLatestMinorOfMajor(major: String): String {
+    return findMatchingVersionNewerThan(Regex("(?<=BUILD-)$major(\\.\\d+)?(?=.txt)"), major) ?: major
+  }
+
+  private fun findEapWithBuildNumberHigherThan(latestEapBuildNumber: String): String? {
```

Comment:
I wouldn't call the param `latestEap...`, as at the moment of running the method, this might no longer correspond to the latest EAP. In fact this method shouldn't care about what this build number represents:

```suggestion
  private fun findEapWithBuildNumberHigherThan(buildNumber: String): String? {
```

It just looks for EAP with a build number higher than the given threshold, whatever this threshold might be